substrate/utils/wasm-builder/src/lib.rs

```diff
@@ -411,9 +380,7 @@ fn get_bool_environment_variable(name: &str) -> Option<bool> {
 
 /// Returns whether we need to also compile the standard library when compiling the runtime.
 fn build_std_required() -> bool {
-	let default = runtime_target() == RuntimeTarget::Wasm;
-
-	crate::get_bool_environment_variable(crate::WASM_BUILD_STD).unwrap_or(default)
+	crate::get_bool_environment_variable(crate::WASM_BUILD_STD).unwrap_or(true)
```

Comment:
Let's add a comment here why.

```suggestion
    // We always rebuild std except if specifically disabled. For Wasm this is necessary because
    // the prebuilt std is compiled with unsupported features. For RISC-V there is no prebuilt binaries
    // available in rustup.
	crate::get_bool_environment_variable(crate::WASM_BUILD_STD).unwrap_or(true)
```